std/zip.d

```diff
@@ -410,19 +412,26 @@ final class ZipArchive
      *
      * Returns: array representing the entire archive.
      */
-    void[] build()
+    void[] build() @safe pure
     {
         import std.algorithm.sorting : sort;
+        import std.string : representation;
+
         uint i;
         uint directoryOffset;
 
         if (comment.length > 0xFFFF)
             throw new ZipException("archive comment longer than 65535");
 
+        static auto vals(T)(T a) @trusted
+        {
+            return a.values();
+        }
+
         // Compress each member; compute size
         uint archiveSize = 0;
         uint directorySize = 0;
-        auto directory = _directory.values().sort!((x, y) => x.index < y.index).release;
+        auto directory = vals(_directory).sort!((x, y) => x.index < y.index).release;
```

Comment:
`byValue.array` is @safe - though in this case a ScopedAppender would be cool to have.